formwidgets/RevisionHistory.php

```diff
@@ -214,10 +216,20 @@ public function onDeleteRevisionById()
      */
     public function onDeleteAllRevisionsByModel()
     {
+        $this->ifReadOnlyChecker();
         if($id = $this->model->id) {
             Revision::where('revisionable_id', $id)->delete();
             Flash::success('All changes successfully deleted!');
+            return;
         }
         Flash::warning('This model could not be found!');
     }
+
+    private function ifReadOnlyChecker()
+    {
+        if ($this->readOnly) {
+            Flash::error(Lang::get('samuell.revisions::lang.revision.read_only_error'));
+            return;
```

Comment:
Just returning would not be enough. The code after `$this->readOnlyChecker()` will still be executed.
Maybe throwing an exception could solve this problem?